### DISCUSSION TITLE
Using mapper's old and new names causes searching issues

### DISCUSSION THREAD

**Discussion by ghost** [ppy/osu#30094] [Ideas]
### Type

Game behaviour

### Bug description

Let's take Mismagius/ Blue Dragon as an example. If you search for maps under the mapper Mismagius, his maps under the alias Blue Dragon don't show up and it's the same the other way around. 
The point is, that all of the maps of a mapper have to show up when you search for their name. 
I don't know how that should be done, as my original idea was to just display the newest name of the mapper for each of their maps which would also be beneficial for new players as they often think that those mappers aren't the same, however I guess there is a reason to keep their old names instead of updating them.

### Screenshots or videos

![screenshot172](https://github.com/user-attachments/assets/9a71e959-1fbc-4fa0-8669-54b5df548a81)

![screenshot173](https://github.com/user-attachments/assets/91517dc5-09a3-4fd5-818c-138f3c2a31a1)

![osu_2024-10-01_20-09-15](https://github.com/user-attachments/assets/1c7c431e-51eb-4dee-9da5-bf325478f5a0)
![osu_2024-10-01_20-09-28](https://github.com/user-attachments/assets/495ffb8e-ab2e-4f54-86e1-af5da5de4656)

### Version

Stable: b20241001 Lazer: 2024.906.2-lazer

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17214206/compressed-logs.zip)

**Comment by PercyDan54** [comment]
The og Big Black was made when he had the name Blue Dragon
And the new Big Black set has tagged "blue dragon", so it only worked one way not the other
![image](https://github.com/user-attachments/assets/501a377d-782e-48a5-9038-552831f6665c)

**Comment by ghost** [comment]
> The og Big Black was made when he had the name Blue Dragon And the new Big Black set has tagged "blue dragon", so it only worked one way not the other ![image](https://private-user-images.githubusercontent.com/50285552/372720078-501a377d-782e-48a5-9038-552831f6665c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc4NjU3NzAsIm5iZiI6MTcyNzg2NTQ3MCwicGF0aCI6Ii81MDI4NTU1Mi8zNzI3MjAwNzgtNTAxYTM3N2QtNzgyZS00OGE1LTkwMzgtNTUyODMxZjY2NjVjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDAyVDEwMzc1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3MzhhNGRkODA0NGE3MWRmMjk3Y2JlODQyNWFiNTJiYTIyN2U2YjAyNDE3ZDU1NTliNmViNjRkMDk5ODg4ZmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.-EVQKqdW-f8-IcUzMmuhZWlbMYUvh7kiO7dgKS_fOyo)

Ooooh I see, thank you:)

**Comment by bdach** [comment]
A mapper's entire username history is not stored in the map files. There's only so much the client can do.

If this is to be handled somehow the machinery responsible for performing the username change is going to need to do something to help out.

I'm downgrading this to discussion since I think this is relatively low priority, and it's unclear what should really happen here.
